receiver/awss3receiver/receiver.go

```diff
@@ -98,8 +105,26 @@ func (r *awss3TraceReceiver) receiveBytes(ctx context.Context, key string, data
 			return err
 		}
 	}
+	return r.dataProcessor.processReceivedData(ctx, r, key, data)
+}
+
+type awss3TraceReceiverProcessor struct {
```

Comment:
`processor` is a loaded term, can you use a different name. Since this is unexported, using `tracereceiver` is probably fine